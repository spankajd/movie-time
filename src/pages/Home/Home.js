import React , { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import './Home.scss';
import MovieList from '../../components/MovieList/MovieList';
import MovieDetails from '../../components/MovieDetails/MovieDetails';
import { fetchMovies } from '../../store/actions/movieActions';
import { selectMovies, selectMoviesError, selectMoviesIsLoading } from "../../store/selectors/movieSelectors";
import SearchBox from '../../components/SearchBox/SearchBox';
import SortByComp from '../../components/SortByComp/SortByComp';
import LoaderScreen from '../../components/LoaderScreen/LoaderScreen';
import Header from '../../components/Header/Header';
function Home() {
  const dispatch = useDispatch();
  const movies = useSelector(selectMovies);
  const isLoading = useSelector(selectMoviesIsLoading);
  const error = useSelector(selectMoviesError);
  
  useEffect( () => {
    dispatch(fetchMovies());
  },[dispatch]);
  
  return (
    <div className="Home">
      <Header></Header>
      <div className="container">
        <div className='top-section'>
          <SortByComp />
          <SearchBox movies={movies} />
        </div>
        <div className='movie-wrapper'>
          <MovieList movies={movies}/>
          <MovieDetails ></MovieDetails>
          </div>
      </div>

      {isLoading && <LoaderScreen imgPath="/assets/loader.gif" msg="Hold On!"/> }
      <LoaderScreen imgPath="/assets/loader.gif" msg={error}/>
    </div>
  );
}

export default Home;
