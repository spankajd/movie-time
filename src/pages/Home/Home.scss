.Home {
  display: flex;
  flex-direction: column;
  height: 100vh;
  
  .container {
    flex-grow: 1; 
    display: flex;
    flex-direction: column;
  }

  .top-section {
    display: flex;
    padding: 15px;
    gap: 20px;
    align-items: center;
    background-color: #6e9dcd;
  }

  .movie-wrapper {
    display: flex;
    flex-direction: row;
    width: 100%;
    font-size: 14px;
    flex-grow: 1;
   
  }
  .movie-list {
    border-right: 1px solid #8badce;
  }



  @media (max-width: 768px) {
    .header {
      position: sticky;
    }

    .top-section {
      flex-direction: column-reverse;
      gap: 10px;
      align-items: end;
    }
    .movie-wrapper > .movie-list{
      width: 100%;
    }
    
    .movie-wrapper > .movie-details{
      display: none;
    }
  }
}