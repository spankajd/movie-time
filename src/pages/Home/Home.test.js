import { screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import { renderWithProviders } from "../../utils/test-utlis";
import Home from "./Home";

describe("Home Component", () => {
    it("renders Rating", () => {
      renderWithProviders(<Home />);
      const element = screen.getByText(/Movie Time/i);
      expect(element).toBeInTheDocument();
    });
    it("matches snapshot", () => {
        const { asFragment } = renderWithProviders(<Home />);
        expect(asFragment()).toMatchSnapshot();
    });
});