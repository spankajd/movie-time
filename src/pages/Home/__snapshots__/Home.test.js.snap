// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <div
      class="header"
    >
      <span
        class="title"
      >
        Movie Time
      </span>
    </div>
    <div
      class="container"
    >
      <div
        class="top-section"
      >
        <div
          class="SortByComp"
        >
          <label
            for="sortBy"
          >
            Sort By: 
          </label>
          <select
            data-testid="sortBy"
            id="sortBy"
          >
            <option
              value=""
            >
              Select
            </option>
            <option
              value="episode_id"
            >
              Episode
            </option>
            <option
              value="release_date"
            >
              Release Date
            </option>
            <option
              value="title"
            >
              Title
            </option>
            <option
              value="imdbRating"
            >
              Rating
            </option>
          </select>
        </div>
        <div
          class="SearchBox"
        >
          <input
            data-testid="SearchBox"
            placeholder="Search movie..."
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="movie-wrapper"
      >
        <div
          class="movie-list"
        />
        <div
          class="movie-details"
        >
          <div
            class="message"
          >
            Please select movie from the list
          </div>
        </div>
      </div>
    </div>
    <div
      class="wrapper"
    >
      <img
        src="/assets/loader.gif"
      />
      <div
        class="message"
      >
        Hold On!
      </div>
    </div>
  </div>
</DocumentFragment>
`;
